ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -80,6 +86,57 @@ public static Builder builder() {
   }
 
   public static Transaction readFrom(final RLPInput input) throws RLPException {
+    if (!ExperimentalEIPs.eip1559Enabled) {
+      return readFromLegacy(input);
+    }
+    input.enterList();
+
+    final Builder builder =
+        builder()
+            .nonce(input.readLongScalar())
+            .gasPrice(Wei.of(input.readUInt256Scalar()))
+            .gasLimit(input.readLongScalar())
+            .to(input.readBytes(v -> v.size() == 0 ? null : Address.wrap(v)))
+            .value(Wei.of(input.readUInt256Scalar()))
+            .payload(input.readBytes());
+
+    final Bytes maybeGasPremiumOrV = input.readBytes();
+    final Bytes maybeFeeCapOrR = input.readBytes();
+    final Bytes maybeVOrS = input.readBytes();
+    final BigInteger v, r, s;
+    // if this is the end of the list we are processing a legacy transaction
+    if (input.isEndOfCurrentList()) {
+      v = maybeGasPremiumOrV.toUnsignedBigInteger();
+      r = maybeFeeCapOrR.toUnsignedBigInteger();
+      s = maybeVOrS.toUnsignedBigInteger();
+    }
+    // otherwise this is an EIP-1559 transaction
```

Comment:
Style: move the comment as the first line of the else block to preserve the `} else {` text.

Reply:
Done